pkg/cmd/update/update.go

```diff
@@ -0,0 +1,453 @@
+	var release *github.RepositoryRelease
+	var err error
+	if opts.targetVersion == "" {
+		release, err = getLatestCorectlRelease(githubClient)
+	} else {
+		release, err = getCorectlReleaseByTag(githubClient, opts.targetVersion)
+	}
+	if err != nil {
+		wizard.Abort(err.Error())
+		return err
+	}
+
+	asset, err := getLatestCorectlAsset(release)
+	if err != nil {
+		wizard.Abort(err.Error())
+		return err
+	}
+	log.Debug().Str("current_version", version.Version).Str("remote_version", asset.Version).Msg("comparing versions")
+	if version.Version == asset.Version {
+		wizard.SetCurrentTaskCompletedTitle(fmt.Sprintf("Already running %s release (%v)", opts.targetVersion, version.Version))
+		wizard.Done()
+		return nil
+	} else {
+		wizard.SetCurrentTaskCompletedTitle(fmt.Sprintf("Update available: %v", asset.Version))
+		wizard.Done()
+	}
+
+	out, err := glamour.Render(asset.Changelog, "dark")
+	if err == nil {
+		_, _ = opts.streams.GetOutput().Write([]byte(out))
+	} else {
+		log.Warn().Err(err).Msg("could not render changelog markdown, falling back to plaintext")
+		_, _ = opts.streams.GetOutput().Write([]byte(asset.Changelog))
+	}
+
+	log.Debug().Bool("skipConfirmation", opts.skipConfirmation).Msg("checking params")
+
+	wizard = opts.streams.Wizard("Confirming update", "Confirmation received")
+	defer wizard.Done()
+	if opts.skipConfirmation {
+		wizard.Info("--skip-confirmation is set, continuing with update")
+	} else {
+		if opts.streams.IsInteractive() {
+			confirmation, err := confirmation.GetInput(opts.streams, fmt.Sprintf("Update to %s now?", asset.Version))
+			if err != nil {
+				return fmt.Errorf("could not get confirmation from user: %+v", err)
+			}
+
+			if confirmation {
+				wizard.Info("Update accepted")
+			} else {
+				err = fmt.Errorf("update cancelled by user")
+				wizard.Abort(err.Error())
+				return err
+			}
+		} else {
+			err = fmt.Errorf("non interactive terminal, cannot ask for confirmation")
+			wizard.Abort(err.Error())
+			return err
+		}
+	}
+
+	wizard.SetTask(fmt.Sprintf("Downloading release %s", asset.Version), fmt.Sprintf("Downloaded release %s", asset.Version))
+	data, err := downloadCorectlAsset(asset)
+	if err != nil {
+		wizard.Abort(err.Error())
+		return fmt.Errorf("could not download release %s: %+v", asset.Version, err)
+	}
+
+	opts.streams.CurrentHandler.SetTask(fmt.Sprintf("Decompressing release %s", asset.Version), fmt.Sprintf("Decompressed release %s", asset.Version))
+	var decompressed *tar.Reader
+	decompressed, err = decompressCorectlAssetInMemory(data)
+	if err != nil {
+		wizard.Abort(err.Error())
+		return fmt.Errorf("could not decompress release %s: %+v", asset.Version, err)
+	}
+
+	path := getCurrentExecutablePath()
+
+	tmpFile, err := os.CreateTemp("", fmt.Sprintf("corectl-%s-", asset.Version))
+	if err != nil {
+		err = fmt.Errorf("unable to create temporary file %s: %+v", asset.Version, err)
+		return err
+	}
+	tmpPath, err := os.Readlink(fmt.Sprintf("/proc/self/fd/%d", tmpFile.Fd()))
```

Comment:
tested it on my mac and it doesn't work

Reply:
I'll test it out on mine and find an alternative way

Reply:
<img width="787" alt="Screenshot 2024-10-22 at 13 48 31" src="https://github.com/user-attachments/assets/8138357e-3ddf-497e-9e2f-e4e14b9008b2">
Working now